packages/@vuepress/theme-default/src/client/components/SidebarItems.vue

```diff
@@ -1,8 +1,47 @@
 <script setup lang="ts">
 import SidebarItem from '@theme/SidebarItem.vue'
+import { watch } from 'vue'
+import { useRouter } from 'vue-router'
 import { useSidebarItems } from '../composables'
 
 const sidebarItems = useSidebarItems()
+
+const route = useRouter()
+
+watch(
+  () => route,
+  (router) => {
+    // the sidebarItem dom whose href equals to the current href of the page
+    const activeSidebarItem = document.querySelector(
+      `[href="${router.currentRoute.value.path}${router.currentRoute.value.hash}"]`
```

Comment:
Could it be better to be more specific? For example, use `.sidebar sidebar-item a[href=...]`? There might be some other links like TOC

Reply:
I have optimized this code.